indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,125 @@
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.colors import ColorTuple
+from indico.modules.events.sessions.controllers.management import RHManageSessionsBase, RHManageSessionBase
+from indico.modules.events.sessions.forms import CreateSessionForm
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.sessions.views import WPManageSessions
+from indico.modules.events.sessions.operations import create_session, edit_session
+from indico.modules.events.sessions.util import get_list_of_colors, get_active_sessions
+from indico.web.flask.templating import get_template_module
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+from MaKaC.webinterface.rh.base import RH
+
+
+def _render_sessions_list(event):
+    tpl = get_template_module('events/sessions/management/sessions/_sessions_list.html')
+    return tpl.render_sessions_list(event, get_active_sessions(event))
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        return WPManageSessions.render_template('management/sessions/sessions_list.html', self.event_new.as_legacy,
+                                                event=self.event_new, sessions=get_active_sessions(self.event_new),
+                                                default_colors=get_list_of_colors())
+
+
+class RHAddSession(RHManageSessionsBase):
+    """Add a session to the event"""
+
+    def _process(self):
+        form = CreateSessionForm()
+        if form.validate_on_submit():
+            event_session = create_session(self.event_new, form.data)
+            db.session.add(event_session)
+            return jsonify_data(sessions_list=_render_sessions_list(self.event_new))
+        return jsonify_template('events/sessions/management/sessions/add_session.html', form=form)
+
+
+class RHEditSession(RHManageSessionBase):
+    """Edit a session"""
+
+    def _process(self):
+        form = CreateSessionForm(obj=self._get_form_defaults())
+        if form.validate_on_submit():
+            edit_session(self.session, form.data)
+            return jsonify_data(sessions_list=_render_sessions_list(self.event_new))
+        return jsonify_template('events/sessions/management/sessions/add_session.html', form=form)
+
+    def _get_form_defaults(self):
+        return FormDefaults(self.session, colors={'background': self.session.background_color,
+                                                  'text': self.session.text_color})
+
+
+class RHRemoveSession(RHManageSessionBase):
+    """Remove a single session"""
+
+    def _process(self):
+        return RH._process(self)
+
+    def _process_DELETE(self):
+        self.session.is_deleted = True
```

Comment:
`delete_session(self.session)` since you'll need to do more than just marking it as deleted (you disassociate contributions etc)